source/slang/hlsl.meta.slang

```diff
@@ -8746,6 +8746,24 @@ __intrinsic_op($(kIROp_MetalAtomicCast))
 [require(metal)]
 T* __getMetalAtomicRef(__ref T x);
 
+// Checks if input is a ImageSubscript
+__generic<T>
+__intrinsic_op($(kIROp_IsTextureAccess))
+[require(metal)]
```

Comment:
`kIROp_IsTextureAccess` works with all targets.
`kIROp_IsTextureAccess` was limited to Metal (at least for now) since this compiler intrinsic only runs a legalization pass if the target is Metal. This was done since (as of this current moment) only Metal uses the intrinsic.

Reply:
I know, but since this is a general mechanism, let's mark the intrinsic as platform neutral.